### ISSUE TITLE
Run accessibility test plan another time before launch

### ISSUE THREAD

**Issue by devinreams** [mozilla-lockbox/lockbox-android#225]
Reminder to run through the test plan first run and documented at #44 again before launch.
### Todo
- [x] AccessibilityScanner
- [x] TalkBack

### Results
  - https://docs.google.com/spreadsheets/d/18WaSUdrcZYiBfW1kgedjaXBdxldzHrgM7dacJE7U-tE/edit?usp=sharing
  - https://github.com/mozilla-lockbox/lockbox-android/issues/493

**Comment by eliserichards** [comment]
(see #223 for previous TalkBack findings)

The reading of on-screen interface items and objects. Testing for:
- All on-screen navigation and button titles are read
- All on-screen alerts and popovers (interrupting the interface) are read
- All labels and text elements on screen are read
- Interaction elements like input boxes and filters have meaningful instructions
- Links to open websites are clearly indicated and read
- Swiping left/right to select next/previous elements works usefully
- Interface scrolling is available where needed

### Welcome Screen (first run)
- [ ] App Name: “Lockbox Firefox Lockbox” is read
- [ ] Tagline: “Take your passwords everywhere” is read
- [ ] “Get Started button” is read — “double tap to activate”
- [ ] “To use Firefox Lockbox...” is read
- [ ] “Learn more button” is read — “double tap to activate”

### Sign in to Firefox Accounts (web view: email)

- [ ] When opened, no default interface item is selected/read
- [ ] Button “X” to close is “unlabeled button”
- [ ] “Enter your email to continue to Firefox Lockbox...” is read
- [ ] “Edit box - Email” is read
- [ ] Keyboard does not appear when input selected (had to hit “Continue” for it to appear, but works OK on “Password” input..)
- [ ] “Valid email required” error is read
- [ ] “Editing edit box: Email” is read
- [ ] “Continue” is read (:warning: not described as a button)

### Sign in to Firefox Accounts (web view: enter password)

- [ ] When loaded, interface is selected and read
- [ ] “Password edit box: Password” is read
- [ ] Double tap on “Password” does show keyboard (unlike email, note above)
- [ ] “Keyboard hidden” is read upon submit

### Sign in to Firefox Accounts (web view: enter password second run)

- [ ] When loaded interface is not selected and read
- [ ] Double tap on “Password” does not show keyboard (unlike above)

### Sign in to Firefox Accounts (web view: confirm this sign in)

- [ ] When loaded, interface is not selected nor read
- [ ] When confirmed, interface is not selected nor read, transition is not announced
- [ ] “Continue button” is read — “double tap to activate”

### List View

- [ ] When loaded, “List” is selected and read (first item could probably be the “Lockbox All Entries A to Z” filter instead)
- [ ] Drawer is “unlabeled button”
- [ ] “All Entries A to Z” is read (not announced its a filter)
- [ ] “Alphabetically” is read - “double tap to activate” :warning” (doesn’t match label of “All Entries A to Z”? not useful to explain it will filter?)
- [ ] Double tap “Alphabetically” and no transition or explanation its already selected/used
- [ ] Unclear if possible to tap or close the popup filter
- [ ] “All Entries Recent” is read (useful title for the filtered view?)
- [ ] Search field announced as “button” (not descriptive)
- [ ] “Search your entries edit box” is read and keyboard activated
- [ ] “Button search clear text button” (clear button description is complex)
- [ ] Each list item can be navigated using left/right swipes
- [ ] Each list item is read with website and username
- [ ] “Double tap to activate” is read on each item
- [ ] When returning from detail view, filtered text edit box is read but the context (like “accounts”) is not read

## Detail view

- [ ] When loaded the web address (https://) is selected and read first (could be the nav bar title instead)
- [ ] Back arrow is “unlabeled button”
- [ ] Web address is read: “https://accounts.firefox.com edit box web address”
- [ ] Username is read: “faketester3 edit box username”
- [ ] Double tap on field does not copy to clipboard (expected?)
- [ ] Double tap copy buttons are read
- [ ] “Username copied” and “Password copied” alerts are read
- [ ] Password is read: “password 21 characters edit box password”
- [ ] Web address, Username, and Password, including revealed, all say “edit box” and “double tap to enter text” (fields not actually editable)
- [ ] “Display your password in plaintext button” is read
- [ ] Double tap reveal button and password field is not selected/re-read nor anything announced
- [ ] After revealed, “display your password in plaintext button” label does not change to opposite meaning (hide)
- [ ] “Learn how to edit this entry” link is read

## Navigation drawer

- [ ] Once loaded, “Menu, Firefox Lockbox” is selected and read
- [ ] All list items can be navigated with left/right gesture
- [ ] “double tap to activate” is read for each labeled item

## Welcome Screen (locked)

- [ ] Once loaded, nothing is selected nor read
- [ ] “Unlock Firefox Lockbox” button is read
- [ ] Android PIN/fingerprint screens are read

## Account Screen

- [ ] Once loaded, back button (not nav bar title) is selected and read
- [ ] Back “unlabeled” button
- [ ] “Firefox Account profile image” is read
- [ ] “Lockbox Tester” (real name) is read
- [ ] Text on screen is read
- [ ] Disconnect button “double tap to activate” is read
- [ ] Popover text is read
- [ ] Button to cancel and disconnect is read

## Settings Screen

- [ ] Once loaded, back button (not nav bar title) is selected and read
- [ ] “Security..” heading is read (but says 7 items because it counts Unlock + Setting + Autofill + Setting)
- [ ] “Unlock with fingerprint” is read
- [ ] Unlock “off switch” setting is not specifically announced as being for that setting
- [ ] “Auto Lock 5 minutes” (setting and value) is read together
- [ ] No announcement to change the auto lock setting  
- [ ] No ability to double tap and change auto lock setting 
- [ ] “Autofill.. Let Firefox ...” setting is read
- [ ] Autofill “off switch” setting is not specifically announced as being for that setting
- [ ] “Send usage data... Mozilla strives” is read
- [ ] Usage data “on switch, double tap to toggle” is not specifically announced as being for that setting
- [ ] “Learn more button” is next after the usage switch so disconnected from the paragraph its related to
- [ ] “App Version” is read

**Comment by eliserichards** [comment]
No breaking accessibility issues found. The most important ones IMO would be: 
- the lack of initial TalkBack description on the onboarding screens
- the "Go to settings" button size with large font (captured in #493)
